src/Enum.php

```diff
@@ -61,16 +68,17 @@ public function __construct($value)
             $value = $value->getValue();
         }
 
-        if (!$this->isValid($value)) {
+        if (false === ($key = static::search($value))) {
             throw new \UnexpectedValueException("Value '$value' is not part of the enum " . static::class);
         }
 
         /** @psalm-var T */
         $this->value = $value;
+        $this->key = $key;
     }
 
     /**
-     * @psalm-pure
+     * @psalm-mutation-free
```

Comment:
I see you changed all those annotations, could you explain?

Reply:
Well `@psalm-pure` is meant for pure methods/functions that have their output dependent only on the input.
Based on this: https://psalm.dev/articles/immutability-and-beyond#mutation-free-methods
`@psalm-mutation-free` is appropriate for getters.

Reply:
I decided not to change the annotations here as a new analysis on them should be done after #134 will be merged
Having them in a separate PR is also better